modules/server/src/network/index.ts

```diff
@@ -1,10 +1,121 @@
-import { NetworkAggregationInterface } from '@/config/types';
+import {
+	buildClientSchema,
+	getIntrospectionQuery,
+	introspectionFromSchema,
+	IntrospectionQuery,
+} from 'graphql';
+import { NetworkAggregationError } from './errors';
+import { fetchGql } from './gql';
+import { NetworkAggregationConfig, NetworkAggregationConfigInput } from './types';
 
-export const createSchemaFromNetworkConfig = ({
-	networkConfig,
+/**
+ * WARNING: Unsafe type check, basically a type assertion. Do not export please.
+ *
+ * This is doing a cursory check that the response from a GQL Server appears to be a IntrospectionQuery object.
+ * If this check passes we will optimistically treat the object as a GQL IntrospectionQuery.
+ *
+ * Should be used sparingly, and ideally all downstream consumers of the object operate within try/catch blocks.
+ *
+ * @param input
+ * @returns
+ */
+const isGqlIntrospectionQuery = (input: unknown): input is IntrospectionQuery =>
+	!!input && typeof input === 'object' && '__schema' in input && typeof input.__schema === 'object';
```

Comment:
wouldn't `typeof input === 'object'` inherently cover the `!!`?
(an `object` cannot be falsy)

```suggestion
	typeof input === 'object' && '__schema' in input && typeof input.__schema === 'object';
```

Reply:
does it matter that this function would allow `input` to be an array customised to include `__schema`?

Reply:
1st point: 
`typeof null` is object - <3 js

Reply:
2nd point:
I don't think that's a concern. I'm using the `getIntrospectionQuery` which is part of the graphql library (conforming to spec) and this check is only happening to check response data from a GQL server.

It queries an automatically generated field in a graphql server, not a user created field.
For an array to occur the graphql spec would need to be changed.

Reply:
is there a `null._schema`? otherwise, `!!` is still redundant